aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/DeploymentSteps.java

```diff
@@ -139,6 +143,74 @@ SdkIterable<GroupNameAndArn> getThingGroupIterable(String coreThingName) {
         return lifecycle.listThingGroupsForAThing(coreThingName).thingGroups();
     }
 
+    /**
+     * implemented the step of installing a custom component with configuration. This requires Recipe for the custom
+     * compoenent.
+     *
+     * @param componentName         the name of the custom component
+     * @param configurationTable    the table which describes the configurations
+     * @throws InterruptedException InterruptedException could be throw out during the component deployment
+     * @throws IOException          IOException could be throw out during preparation of the CLI command
+     */
+    @When("I install the component {word} from local store with configuration")
+    public void installComponentWithConfiguration(final String componentName, final String configurationTable)
+            throws InterruptedException, IOException {
+        // read the recipe from local store, and get component name and version from recipe
+        List<String> componentSpecs = Arrays.asList(
+                componentName, LOCAL_STORE_RECIPES.resolve(String.format("%s.yaml", componentName)).toString()
+        );
+
+        // handle the config input
+        List<Map<String, Object>> configuration = readConfigurationTable(configurationTable);
```

Comment:
Nit: What we're doing here is just reading configuration for one single component. So this can be a map instead of list of maps. Also rename this to `readConfiguration` as the config we're passing in steps is not a table but string.

Reply:
I will renamed it. But, The reason of the list of maps is to re-use existing classes and methods of OTF

Reply:
List of maps is only used in the internal methods of the newly added code for reading configuration. It can be modified.